rest/src/main/scala/io/udash/rest/SttpRestClient.scala

```diff
@@ -30,14 +28,9 @@ object SttpRestClient {
     asHandleRequest(uri"$baseUri")
 
   private def toSttpRequest(baseUri: Uri, request: RestRequest): Request[Array[Byte], Nothing] = {
-    val uri = baseUri |>
-      (u => u.copy(path = u.path ++
-        request.parameters.path.map(_.value))) |>
-      (u => u.copy(queryFragments = u.queryFragments ++
-        request.parameters.query.entries.iterator.map {
-          case (k, PlainValue(v)) => KeyValue(k, v, QueryFragmentEncoding.All, QueryFragmentEncoding.All)
-        }.toList
-      ))
+    val tmpUri = uri"${request.parameters.toUri(s"tmp://tmp")}"
+    val uri = baseUri |> (u =>
+      u.copy(path = u.path ++ tmpUri.path, queryFragments = u.queryFragments ++ tmpUri.queryFragments))
```

Comment:
Not tested well enough.
```scala
      u.copy(path = u.path ++ tmpUri.path, queryFragments = tmpUri.queryFragments))
```
passes the tests.

Reply:
I actually wonder how much should I care about it. Maybe I should just reject base URI with query parameters.

Reply:
Well, someone might need this obscure feature to pass a parameter along all requests. It wouldn't be easy to do with the framework otherwise I guess.